listings/applications/components/src/ownable.cairo

```diff
@@ -93,3 +98,159 @@ pub mod ownable_component {
+        let (ownable, _) = deploy();
+
+        assert(ownable.owner() == owner, 'wrong_owner');
+    }
+
+    #[test]
+    fn test_transfer_ownership_by_owner() {
+        set_contract_address(contract_address_const::<'initial'>());
+        let (ownable, _) = deploy();
+        let owner = contract_address_const::<'owner'>();
+        ownable.transfer_ownership(owner);
+
+        assert(ownable.owner() == owner, 'wrong_owner');
+    }
+
+    #[test]
+    #[should_panic]
+    fn test_transfer_ownership_not_owner() {
+        set_contract_address(contract_address_const::<'initial'>());
+        let (ownable, _) = deploy();
+
+        set_contract_address(contract_address_const::<'not_owner'>());
+        ownable.transfer_ownership(contract_address_const::<'new_owner'>());
+    }
+
+    #[test]
+    #[should_panic]
+    fn test_transfer_ownership_account_zero() {
+        set_contract_address(contract_address_const::<'initial'>());
+        let (ownable, _) = deploy();
+
+        ownable.transfer_ownership(Zero::zero());
+    }
+
+    #[test]
+    #[should_panic]
+    fn test_transfer_ownership_by_zero() {
+        set_contract_address(Zero::zero());
+        let (ownable, _) = deploy();
+
+        ownable.transfer_ownership(contract_address_const::<'new_owner'>());
+    }
+
+    #[test]
+    fn test_renounce_ownership() {
+        set_contract_address(contract_address_const::<'owner'>());
+        let (ownable, _) = deploy();
+
+        ownable.renounce_ownership();
+        assert(ownable.owner() == Zero::zero(), 'not_zero_owner');
+    }
+
+    #[test]
+    #[should_panic]
+    fn test_renounce_ownership_previous_owner() {
+        set_contract_address(contract_address_const::<'owner'>());
+        let (ownable, _) = deploy();
+
+        ownable.renounce_ownership();
+        ownable.transfer_ownership(contract_address_const::<'new_owner'>());
+    }
+
+    #[test]
+    #[should_panic]
+    fn test_renounce_ownership_not_owner() {
+        set_contract_address(contract_address_const::<'owner'>());
+        let (ownable, _) = deploy();
+
+        set_contract_address(contract_address_const::<'not_owner'>());
+        ownable.renounce_ownership();
+    }
+
+    #[test]
+    fn test_ownership_renounced_event(){
+        let contract_address = contract_address_const::<'owner'>();
+        set_contract_address(contract_address);
+        let (ownable, address) = deploy();
+        ownable.renounce_ownership();
+        set_contract_address(address);
+        assert_eq!(
+            starknet::testing::pop_log(contract_address),
+            Option::Some(Event::OwnershipRenouncedEvent(OwnershipRenouncedEvent { previous:contract_address }))
+        );
+    }
+
+    #[test]
+    fn test_ownership_transferred_event(){
+        let contract_address = contract_address_const::<'owner'>();
+        set_contract_address(contract_address);
+        let (ownable, address) = deploy();
+        let owner = contract_address_const::<'owner'>();
```

Comment:
```suggestion
        let new_owner = contract_address_const::<'new_owner'>();
```
'owner' is the original owner, so the test is transferring ownership from an owner to themselves, which would make little sense.